server/routes/induction/update/previousWorkExperienceTypesUpdateController.ts

```diff
@@ -0,0 +1,251 @@
+      ]
+    }
+    const { previousWorkExperienceTypesForm } = req.session
+
+    const errors = validatePreviousWorkExperienceTypesForm(previousWorkExperienceTypesForm, prisonerSummary)
+    if (errors.length > 0) {
+      req.flash('errors', errors)
+      return res.redirect(`/prisoners/${prisonNumber}/induction/previous-work-experience`)
+    }
+
+    if (
+      noChangesToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm) ||
+      onlyRemovalsWereMadeToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm)
+    ) {
+      if (onlyRemovalsWereMadeToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm)) {
+        logger.debug(
+          'Previous Work Experiences changes are only removed types so Induction can be updated without starting a flow',
+        )
+        // call the service to update the Induction
+        try {
+          await this.updateInduction(inductionDto, previousWorkExperienceTypesForm, req, prisonerSummary)
+        } catch (e) {
+          return next(createError(500, `Error updating Induction for prisoner ${prisonNumber}. Error: ${e}`))
+        }
+      } else {
+        logger.debug('No changes to Previous Work Experiences were submitted')
+      }
+
+      req.session.previousWorkExperienceTypesForm = undefined
+      req.session.inductionDto = undefined
+      return res.redirect(`/plan/${prisonNumber}/view/work-and-interests`)
+    }
+
+    /* We need to show the Details page for each of:
+         - previous work experience on the original induction that has not been removed by the form submission
+         - any additional job types that have been added by the form submission
+         - and OTHER if it's value has changed
+     */
+    const workExperienceTypesToShowDetailsFormFor = [
+      ...typesOfPreviousWorkExperienceToShowDetailsFormFor(inductionDto, previousWorkExperienceTypesForm),
+    ].sort(previousWorkExperienceTypeScreenOrderComparator) // sort them by the order presented on screen (which is not alphabetic on the enum values)
+
+    logger.debug(
+      `Previous Work Experiences changes resulting in going to the Detail pages for ${workExperienceTypesToShowDetailsFormFor}`,
+    )
+
+    // TODO - replace with correct redirect once we've implemented form redirect/handling for a "queue" of Previous Work Experience Detail pages
+    return res.redirect(`/plan/${prisonNumber}/view/work-and-interests`)
+  }
+
+  /**
+   * Updates the Induction using the InductionService.
+   * @throws Error if InductionService throws an error. The Error from InductionService is rethrown.
+   */
+  private updateInduction = async (
+    inductionDto: InductionDto,
+    previousWorkExperienceTypesForm: PreviousWorkExperienceTypesForm,
+    req: Request,
+    prisonerSummary: PrisonerSummary,
+  ) => {
+    const updatedInduction = updatedInductionDtoWithRemovedPreviousWorkExperiencesAndChangesToOther(
```

Comment:
Nice descriptive method name! 😂 😉 

In all seriousness, I'm wondering if this a bit too involved and could be simplified? In terms of work experience types, the submitted form is the "final source of truth". It just lacks the job details. Would it not be simpler/easier to just build up a list of job types and extract the missing roles from the existing ones in the session?

I did something similar in my PR: https://github.com/ministryofjustice/hmpps-education-and-work-plan-ui/pull/461/files#r1510958161

Reply:
Not sure I follow ... lets chat... 👍